website/docs/developer-section/quickstart/create-verifiable-document-issuer.md

```diff
@@ -127,840 +123,229 @@ When you reload the app, the MetaMask extension should prompt you for your passw
 
 ```tsx
+import { JsonRpcSigner } from "@ethersproject/providers";
+import { DocumentStoreFactory } from "@govtechsg/document-store";
+
 export const deployDocumentStore = async (signer: JsonRpcSigner) => {
   const factory = new DocumentStoreFactory(signer);
-  const documentStore = await factory.deploy("DEMO_DOCUMENT_STORE");
+  const documentStore = await factory.deploy("My Document Store", await signer.getAddress());
   await documentStore.deployTransaction.wait();
   return documentStore.address;
 };
 ```
 
-This function deploys a document store from a `DocumentStoreFactory` and returns the address of the deployed document store. Typically, once the document store is deployed, we can save this address in a persistent storage and reuse it whenever we run the application. In order to keep things light-weight however, we will simply want to store this address in state.
+This function deploys a Document Store from a `DocumentStoreFactory` and returns the address of the deployed Document Store. Typically, once the Document Store is deployed, we can save this address in a persistent storage and reuse it whenever we run the application. In order to keep things light-weight however, we will simply want to store this address in state.
 
-To do so, we will create a file `AppContext.tsx`
+For convenience sake, we will create a file `AppContext.tsx` to house all the "global" states, while `AccountContext.tsx` for all metamask related states. You should look into other state management tools once your application scales up.
 
 ```tsx
-import { JsonRpcSigner } from "@ethersproject/providers";
 import { createContext } from "react";
+import { documentStoreAddress } from "../types";
 
-interface IAppContext {
-  signer?: JsonRpcSigner;
-  documentStoreAddress?: string;
-  setDocumentStoreAddress: (documentStoreAddress: string) => void;
-}
-
-export const AppContext = createContext<IAppContext>({
+export const AppContext = createContext<{
+  documentStoreAddress: documentStoreAddress;
+  setDocumentStoreAddress: (documentStoreAddress: documentStoreAddress) => void;
+}>({
+  documentStoreAddress: null,
   setDocumentStoreAddress: () => null,
 });
 ```
 
-We can import `AppContext` into `App.tsx` so that the next few components we create have easy access to the `signer`, `documentStoreAddress` and any other values they might need.
+```tsx
+import { JsonRpcSigner } from "@ethersproject/providers";
+import { createContext } from "react";
+import { signer, network } from "../types";
+
+export const AccountContext = createContext<{
+  signer: signer;
+  setSigner: (signer: JsonRpcSigner) => void;
+}>({
+  signer: null,
+  setSigner: () => null,
+});
+```
+
+We can import `AppContext` and `AccountContext.tsx` into `App.tsx` so that the next few components we create have easy access to the `signer`, `documentStoreAddress` and any other values they might need.
```

Comment:
u up for challenge to compose the contexts together?

https://javascript.plainenglish.io/how-to-combine-context-providers-for-cleaner-react-code-9ed24f20225e

Reply:
thanks for the tip https://github.com/Open-Attestation/demo-verifiable-document-issuer/pull/1